src/lib/core/TLVReader.cpp

```diff
@@ -330,6 +331,25 @@ CHIP_ERROR TLVReader::Get(CharSpan & v) const
     }
 
     v = CharSpan(Uint8::to_const_char(bytes), len);
+#if CHIP_CONFIG_TLV_VALIDATE_CHAR_STRING_ON_READ
+    // Spec requirement: A.11.2. UTF-8 and Octet Strings
+    //
+    // For UTF-8 strings, the value octets SHALL encode a valid
+    // UTF-8 character (code points) sequence.
+    //
+    // Senders SHALL NOT include a terminating null character to
+    // mark the end of a string.
+
+    if (!Utf8::IsValid(v))
+    {
+        return CHIP_ERROR_INVALID_UTF8;
+    }
+
+    if ((len > 0) && (v[len - 1] == 0))
```

Comment:
Please use v directly:
`if (!v.empty() && (v[v.size() - 1] == 0))`

Reply:
Also, we should have a 

```
bool HasTrailingNull(CharSpan str)
{
  return !str.empty() && (str[str.size() - 1] != 0);
}
```

and put it in Span.h, that way the code can be reused, rather than needing to inline it here and in the TLVWriter.

Reply:
Why? my logic for the above was that it is less characters.

Reply:
It's indirect --> if changes get done, this can start failing to check-out. There are no tests to keep it honest.

Reply:
Ok, used v only without len.